torchvision/datasets/imagenet.py

```diff
@@ -21,6 +21,11 @@
 class ImageNet(ImageFolder):
     """`ImageNet <http://image-net.org/>`_ 2012 Classification Dataset.
 
+    Before using this class, it is required to download `ImageNet 2012` dataset from
+    `here <https://image-net.org/challenges/LSVRC/2012/2012-downloads.php>`_ and
+    place ILSVRC2012_devkit_t12.tar.gz, ILSVRC2012_img_train.tar or
+    ILSVRC2012_img_val.tar based on split in the root directory.
```

Comment:
Nits:

1. Wrap the paragraph into a note directive
2. Wrap the file names in double backticks to "separate" them from the normal text

```suggestion
    .. note::
        Before using this class, it is required to download `ImageNet 2012` dataset from
        `here <https://image-net.org/challenges/LSVRC/2012/2012-downloads.php>`_ and
        place the files ``ILSVRC2012_devkit_t12.tar.gz`` and ``ILSVRC2012_img_train.tar``
        or ``ILSVRC2012_img_val.tar`` based on ``split`` in the root directory.
```